modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+Linux Integration (64 bit)::
+Available in Linux. This provider uses a random value and relies on the `libsecret` library that must be present and working on your Linux distribution. This provider uses the master password automatically to decrypt your credentials.
+UI Prompt::
+Available in Windows, macOS, and Linux. This provider prompts you to create a master password and uses it to encrypt your credentials. This provider requires you to enter your master password every time you open Studio.
+
+When adding a new credential for the first time, Studio notifies you that a new master password has been created. Additionally, Studio prompts you to create two recovery questions to use in case you forget your master password. You must treat the security questions as a second password. If you lose your master password, your security questions are the only other method to recover your credentials.
+
+Each provider supplies its own master password. When you use a provider to encrypt a new credential, you can only use that provider to recover the encrypted credential.
+
+Once a master password is created, Studio caches it until it is closed, or you manually clear the password from Studio's cache.
+
+== Secure Storage File
+
+This file holds all your user credentials in an encrypted form. By default, the secure storage file is located in `$HOME/.eclipse/org.eclipse.equinox.security/secure_storage`.
+
+Because it is a single file, the same secure storage is shared across all Eclipse-based workspaces and Anypoint Studio instances.
+
+
+== Update the Master Password
+
+You can change or rotate the master password for a particular provider. When changing a master password, all the encrypted data by this provider is re-encrypted with the new master password.
+
+=== Before you Begin
+
+* Ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
+Because the secure storage file is shared across all running instances and the master password is cached in each instance's memory, updating your master password causes other opened instances to still write the credentials using the old master password. Closing all other instances ensures you avoid the <<no-password-provided-exception,No Password Provided Exception>>.
+
+To re-generate your master password:
+
+. Navigate to *Preferences* > *General* > *Security* > *Secure Storage*.
+. Select *Change Password...*:
++
+image::secure-storage-change-password.png[]
+. Studio prompts you to create a password hint for this new password. +
+Select *Yes* to generate a new password hint:
++
+image::new-password-created.png[]
++
+[CAUTION]
+--
+If you don't supply new password hints, then the former ones persist, which reference the former password. You cannot recover this new password using the previous password hints.
```

Comment:
The usage of "password hint" it is really misleading from Eclipse side, they are really "Recovery questions/answers". I understand that this is to be close to UI messages but maybe it could be clarified.

Also, on each reference to "password" of this block I would make explicitly that it is the master password that it is going to be changed.

Reply:
Good point. I want to refer to them as "password hints" because of the UI. 
We always documented Studio as if the user is not familiar with Eclipse, so it's preferable to stick with the definitions that are presented to the user in Studio. 
In the first mention of "password hint" I'll add a clarification that these are "security questions".